### DISCUSSION TITLE
Security Issue? Ubuntu firewall settings for RuskDesk client on LAN

### DISCUSSION THREAD

**Discussion by dr1ver1** [rustdesk/rustdesk#4486] [Q&A]
To access the RustDesk client on Ubuntu from a RustDesk client on another machine on the LAN (going through a RustDesk Server on the LAN), I needed to add the following to the firewall on the machine being accessed:
`sudo ufw allow from 192.168.1.0/24 proto tcp`
i.e., by default TCP connections from other clients on the LAN are blocked with a vanilla Ubuntu & UFW set up. The RustDesk client from the requesting machine needs TCP to be allowed, and the ports seem to be "random" - at least they are not 21115-21119, and seem to vary each time. This was evidenced in `/var/log/ufw.log`

Maybe I missed it, but I haven't this in any of the documentation - note this is for the client not the RustDesk server.
Maybe it is obvious, but I spent some time on discord and no one there picked it up, and I haven't seen it mentioned in any of the issues or discussions here. I suspect there may be some potential users who have the same issue and have given up, especially if they are not familiar with looking at firewall logs.

I'd like to know if there is a better solution, as I don't like having all ports open even though it is restricted to the LAN.

I haven't add a chance to investigate how this works when the requesting client is outside of the LAN, because then 192.168.1.0/24 is no longer relevant.

Ref: #3951

**Comment by dr1ver1** [comment]
This seems to be a significant security issue: you need to open all ports on the firewall as the RustDesk client is initially communicating on a "random"(?) port. What am I missing? How are other people handling this?

**Comment by jimisens** [comment]
I'm currently using RustDesk to offer a maintenance service to a client with multiple stores. My client has a firewall that controls computers' Internet access.
We can access certain stores remotely without any problem but there is no way to others.
My client asked me what ports he should open and I gave him the ones announced by Rustdesk.
TCP (21114, 21115, 21116, 21117, 21118, 21119)
UDP (21116)

However, even opening these ports in the firewall, we still cannot connect to the computers.

I think the problem lies precisely in what is mentioned here, that the Rustdesk client uses random ports and these are blocked by the firewall.
For me it is important to be able to define the range of ports used, so that my client can open them in a standard way in their firewall. 
The idea of ​​opening all ports will never be accepted!!
Does anyone have a solution?

**Comment by cdkagaya** [comment]
I also meet the same issue, I must close the firewall to use it. As the port used by client side is random, it is difficult to config..
Is there any update?

**Comment by duckarp** [comment]
Any update on this issue?

**Reply by wangxihui1** [reply]
I am afraid nobody care this big issue, cannot understand

**Reply by duckarp** [reply]
I suppose only "solution" then is do something like this:
https://github.com/rustdesk/rustdesk/issues/3951#issuecomment-1562998236

Like this you can at least have the firewall enabled...

**Comment by flier268** [comment]
I post a issue #13656
